keygen/src/keygen.rs

```diff
@@ -68,16 +71,16 @@ fn get_keypair_from_matches(
     config: Config,
     wallet_manager: &mut Option<Rc<RemoteWalletManager>>,
 ) -> Result<Box<dyn Signer>, Box<dyn error::Error>> {
-    let mut path = dirs_next::home_dir().expect("home directory");
-    let path = if matches.is_present("keypair") {
-        matches.value_of("keypair").unwrap()
+    let source = if matches.try_contains_id("keypair")? {
+        Cow::Borrowed(matches.get_one::<SignerSource>("keypair").unwrap())
     } else if !config.keypair_path.is_empty() {
-        &config.keypair_path
+        Cow::Owned(SignerSource::parse(&config.keypair_path)?)
     } else {
+        let mut path = dirs_next::home_dir().expect("home directory");
         path.extend([".config", "solana", "id.json"]);
-        path.to_str().unwrap()
+        Cow::Owned(SignerSource::parse(path.to_str().unwrap())?)
     };
```

Comment:
Not obvious to me: why is the `Cow` wrapping necessary? If this will be a common need, it's unfortunate ergonomics.

Reply:
Yeah, this is unfortunate ergonomics. The issue is that `matches.get_one::<SignerSource>("keypair").unwrap()` gives a reference `&SignerSource` while `SignerSource::parse(...).unwrap()` gives a fully owned `SignerSource`. Returning `&SignerSource::parse(...).unwrap()` is not possible since the owned type will be freed out of scope.

The alternative would be to define an owned variable outside of the if statement, set it to `SignerSource::parse(...).unwrap()`, and return a reference to it. I thought this was less elegant than using `Cow` at first, but it seems like it could be cleaner, so I made the change. Please take a look.

Reply:
Thanks, I agree the current version is a little cleaner.
It's actually odd to me that `ArgMatches::get_one` returns a reference instead of an owned object. I suppose I should read the clap code to see why.